ClickHouse.Client/Formats/BinaryStreamWriter.cs

```diff
@@ -197,7 +197,17 @@ public void Write(EnumType enumType, object value)
             writer.Write(enumIndex);
         }
 
-        public void Write(NestedType nestedType, object value) => throw new NotSupportedException("Writing Nested values directly is not supported, see documentation");
+        public void Write(NestedType nestedType, object value) 
+        {
+            // => throw new NotSupportedException("Writing Nested values directly is not supported, see documentation");
+
+            var items = (object[])value;            
```

Comment:
I would suggest converting to `IEnumerable` here & using `foreach` - should be able to use `List<T>` just as well

Reply:
Why would you convert the data type here? It's essentially the same as the Tuple implementation.

Reply:
Because you are already converting to `object[]`, and it significantly narrows down the types you can pass here. `IEnumerable` would allow to use both `List` and array, for example

Reply:
agreed, it's updated